### DISCUSSION TITLE
Can d.ry copy env vars in `make instance` shell?

### DISCUSSION THREAD

**Discussion by mcmikemn** [EnigmaCurry/d.rymcg.tech#87] [Q&A]
When you `make instance`, d.rymcg.tech creates a sub-shell for the new instance. Can that subshell retain or copy the environment from the main shell? For example, I have aliases set up in .bashrc in my main shell, but they're not available in the sub-shell.

**Comment by mcmikemn** [comment]
If it's possible, I'm guessing something in the `switch` target in Makefile.instance needs to change.
`bash --rcfile <(echo "export TERM=$$TERM; PATH=$$PATH; export INSTANCE=$${instance}; export PROJECT_INSTANCE=$$(basename $${PWD})_$${instance}; export PROJECT=$$(basename $$PWD); export INSTANCE_SUBSHELL=true; unset ENV_FILE; unset MAKELEVEL; PS1='\n(context=$$(${BIN}/docker_context) project=$$(basename $$PWD) instance=$${instance})\n\W $$ '") -i)`

Could we add a `source ~/.bashrc`?

Also, I notice that PATH isn't `export`ed, but all the other env vars are. Is that intentional?

**Reply by EnigmaCurry** [reply]
PATH does not need to be exported, it is a shell variable only, not for another app to consume. You can of course manually run `source ~/.bashrc` in the subshell. We could add a source to the inline command, but it may behave weirdly for different users, since we don't control ~/.bashrc - it should be put at the start so that the other vars override the ones from ~/.bashrc - test it, does it work? Also a user may not have a .bashrc so need to test for its existance. What are your aliases that you need? I'm not sure why you'd need any aliases in the subshell?

**Reply by EnigmaCurry** [reply]
Maybe add root config variable in .env-dist like MAKE_INSTANCE_SOURCE_BASHRC=true but it defaults to false, so it only sources bashrc when its true.

**Comment by EnigmaCurry** [comment]
I have converted this discussion into an issue #89 as this is a feature request.

**Comment by mcmikemn** [comment]
> What are your aliases that you need? I'm not sure why you'd need any aliases in the subshell?

I have `alias ls='ls -alh --color'`. I don't need it, I just like it. It's no big deal if I have to manually type the full command in the subshell.

> Maybe add root config variable in .env-dist like MAKE_INSTANCE_SOURCE_BASHRC=true but it defaults to false, so it only sources bashrc when its true.

I like this idea. I'll work on it.

**Reply by EnigmaCurry** [reply]
Cool, see my updated recommendation in #89 to use `MAKE_INSTANCE_SOURCE_RCFILE` instead, to be able to source a file other than bashrc
